### DISCUSSION TITLE
Remove untitled slides from slide-list, but keep them visible

### DISCUSSION THREAD

**Discussion by ellafm** [quarto-dev/quarto-cli#10478] [Q&A] [answered]
### Description

I am making a reveal.js presentation in quarto, and I have some slides that don't have a title. I don't want these to appear in the slide-list that is shown in the navigation window, but I can't figure out how to do it.

I know that reveal.js-menu can be configured to provide titleSelector: 'h1' and hideMissingTitles: false, but I've tried including these in the yaml with no effect.

I also tried setting the data-menu-title to an empty string, but when I compiled the presentation this was over-ridden and a title for the slide was generated.

**Comment by mcanouil** [comment]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by ellafm** [reply]
````{qmd}
---
title: "Some title"
author: "An author"
jupyter: python3
format: 
    revealjs:
        titleSelector: 'h1'
        useTextContentForMissingTitles: false 
        hideMissingTitles: true
        theme: default
        slide-number: true
---

# First section

## Slide with title

##

Slide without a title

````

Here is the output of quarto check:

```txt
Quarto 1.5.56
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.56
      Path: C:\Users\xxxxxxx\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.12.3
      Path: C:/Users/xxxxxxx/AppData/Local/Programs/Python/Python312/python.exe
      Jupyter: 5.7.2
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Reply by mcanouil** [reply]
Why are you setting empty titles if you don't want them?
Use `---` to make a new slide without a title as documented https://quarto.org/docs/presentations/revealjs/

**Reply by ellafm** [reply]
I originally used the approach documented at the link you sent, but it had the same effect, and while I was investigating a solution I switched to using empty titles because I thought I would be able to pass attributes or class options that might fix the issue

**Reply by mcanouil** [reply, marked as answer]
The option you've set are not passed on to the Reveal.js plugin.

You did not place the options in the proper scope.
The options are for the menu plugin, so you need to tell Quarto that.
See <https://quarto.org/docs/presentations/revealjs/presenting.html#navigation-menu> for an example of this.


```qmd
---
title: "Some title"
author: "An author"
jupyter: python3
format: 
  revealjs:
    menu:
      titleSelector: 'h1'
      useTextContentForMissingTitles: false 
      hideMissingTitles: true
    theme: default
    slide-number: true
---

# First section

## Slide with title

##

Slide without a title
```

**Comment by ellafm** [comment]
It works! Oh my you are a clever one. Thank you for the help.
